modules/foreman_compute_profile.py

```
@@ -82,38 +165,76 @@
+        compute_attribute_dict['compute_resource'] = find_compute_resource(module, name=compute_attribute_dict['compute_resource'], failsafe=False)
+    except Exception as e:
+        module.fail_json(msg='Failed to find entity: %s ' % e)
 
-    data = cement.sanitize_entity_dict(data, name_map)
+    if 'compute_profile' not in compute_attribute_dict:
+        compute_attribute_dict['compute_profile'] = compute_profile
 
-    changed = cement.naildown_entity_state(nailgun.entities.ComputeProfile, data, compute_profile, state, module)
+    changed = naildown_entity_state(ComputeAttribute, compute_attribute_dict, compute_attribute, 'present', module)
 
     return changed
 
 
-if __name__ == '__main__':
+def main():
     module = ForemanEntityAnsibleModule(
         argument_spec=dict(
             name=dict(required=True),
             updated_name=dict(),
+            compute_attributes=dict(type='list'),
         ),
         supports_check_mode=True,
     )
-    changed = main(module)
+
+    (compute_profile_dict, state) = module.parse_params()
+    name = compute_profile_dict.get('name')
+    updated_name = compute_profile_dict.get('updated_name')
+    compute_attributes = compute_profile_dict.pop('compute_attributes', list())
+
+    module.connect()
+
+    try:
+        # Try to find the compute_profile to work on
+        compute_profile = find_compute_profile(module, name=compute_profile_dict['name'], failsafe=True)
+    except Exception as e:
+        module.fail_json(msg='Failed to find entity: %s ' % e)
+
+    if state == 'present' and updated_name:
+        compute_profile_dict['name'] = updated_name
+
+    compute_profile_dict = sanitize_entity_dict(compute_profile_dict, name_map)
+
+    (changed, compute_profile) = naildown_entity(ComputeProfile, compute_profile_dict, compute_profile, state, module)
+
+    # Apply changes on underlying compute attributes only when present
```

Can we propagate an error when specifying attributes while deleting the compute-resource?

Compare: https://github.com/theforeman/foreman-ansible-modules/blob/3c1042e038213f5ee9270e2c369d69a811fcef77/modules/foreman_provisioning_template.py#L351

---

I didn't read the provisioning template module before. I will change the code to reflect the same behavior.

---

You added this (?): https://github.com/theforeman/foreman-ansible-modules/pull/243/files#diff-a08444d75f83dc9982ac920da6704a3aR212
That's fine!